docs/faq.rst

```diff
@@ -0,0 +1,211 @@
+
+Look at the ``host`` argument of :meth:`~asyncio.loop.create_server`.
+
+:func:`serve` accepts the same arguments as
+:meth:`~asyncio.loop.create_server`.
+
+How do I close a connection properly?
+.....................................
+
+websockets takes care of closing the connection when the handler exits.
+
+How do I run a HTTP server and WebSocket server on the same port?
+.................................................................
+
+This isn't supported.
+
+Providing a HTTP server is out of scope for websockets. It only aims at
+providing a WebSocket server.
+
+There's limited support for returning HTTP responses with the
+:attr:`~server.WebSocketServerProtocol.process_request` hook.
+If you need more, pick a HTTP server and run it separately.
+
+Client side
+-----------
+
+How do I close a connection properly?
+.....................................
+
+Use :func:`connect` as a context manager and trust websockets::
+
+    async with connect(...) as websocket:
+        ...
+
+How do I reconnect automatically when the connection drops?
+...........................................................
+
+See `issue 414`_.
+
+.. _issue 414: https://github.com/aaugustin/websockets/issues/414
+
+How do I disable SSL certificate verification?
+..............................................
+
+Look at the ``ssl`` argument of :meth:`~asyncio.loop.create_connection`.
+
+:func:`connect` accepts the same arguments as
+:meth:`~asyncio.loop.create_connection`.
+
+Architecture
+------------
+
+How do I do two things in parallel? How do I integrate with another coroutine?
+..............................................................................
+
+You must start two tasks, which the event loop will run concurrently. You can
+achieve this with :func:`asyncio.gather` or :func:`asyncio.wait`.
+
+This is also part of learning asyncio and not specific to websockets.
+
+Keep track of the tasks and make sure they terminate or you cancel them when
+the connection terminates.
+
+How do I create channels or topics?
+...................................
+
+websockets doesn't have built-in publish / subscribe for these use cases.
+
+Depending on the scale of your service, a simple in-memory implementation may
+do the job or you may need an external publish / subscribe component.
+
+Are there ``onopen``, ``onmessage``, ``onerror``, and ``onclose`` callbacks?
+............................................................................
+
+No, there aren't.
+
+websockets provides high-level, coroutine-based APIs. Compared to callbacks,
+coroutines make it easier to manage control flow in concurrent code.
+
+If you prefer callback-based APIs, you should use another library.
+
+Can I use ``websockets`` synchronously, without ``async`` / ``await``?
+......................................................................
+
+You can convert every asynchronous call to a synchronous call by wrapping it
+in ``asyncio.get_event_loop().run_until_complete(...)``.
```

Comment:
I think there is `asyncio.run` now.

Reply:
It won't work here because it always creates a new event loop and closes it at the end.